tools/serverpod_cli/lib/src/config/config.dart

```diff
@@ -283,6 +275,25 @@ class GeneratorConfig {
     );
   }
 
+  static Future<void> _warnIfDependenciesAreNotInstalled(
+      String serverRootDir, Pubspec pubspec) async {
+    var packageConfig = await findPackageConfig(Directory(serverRootDir));
+
+    if (packageConfig == null) {
+      throw const ServerpodProjectNotFoundException(
+        'Failed to read your server\'s package configuration. Have you run '
+        '`dart pub get` in your server directory?',
+      );
+    }
+
+    var allPackagesAreInstalled = pubspec.dependencies.keys
+        .every((dependencyName) => packageConfig[dependencyName] != null);
```

Comment:
Is there a case for checking `dev_dependencies` as well?

Reply:
As of today, no since we only care about modules here and they must be normal dependencies.